tests/fixtures/cloud_uploader.py

```diff
@@ -88,6 +84,16 @@ def upload_file(
             multipart_chunksize=chunk_sz,
         )
 
+        if metadata_keys is None:
```

Comment:
You only need to set one (tags vs metadata_keys), not both.

I personally prefer tags because they are immutable, whereas metadata keys are mutable.

Reply:
Cool that tags are immutable :)

btw, tags and metadata_keys args were never populated simultaneously in `populate_lib.py`, as far as I can tell. The test `test_file_put_metadata_from_tags` passes an empty dict to metadata_keys, which is the design pattern I was following along with. This is not an argument in favor of my change, just a statement of rationale.